core/src/subgraph/runner.rs

```diff
@@ -674,20 +675,32 @@ where
 
     fn create_dynamic_data_sources(
         &mut self,
-        created_data_sources: Vec<DataSourceTemplateInfo<C>>,
+        created_data_sources: Vec<InstanceDSTemplateInfo>,
     ) -> Result<(Vec<DataSource<C>>, Vec<Arc<T::Host>>), Error> {
         let mut data_sources = vec![];
         let mut runtime_hosts = vec![];
 
         for info in created_data_sources {
-            // Try to instantiate a data source from the template
+            let manifest_idx = info
+                .template
+                .manifest_idx()
+                .ok_or_else(|| anyhow!("Expected template to have an idx"))?;
+            let x = self
+                .inputs
+                .templates
+                .iter()
+                .find(|t| t.manifest_idx() == manifest_idx)
+                .ok_or_else(|| {
+                    anyhow!("Expected to find a template for this dynamic data source")
+                })?;
```

Comment:
Could we have a more descriptive name for `x` ?

Reply:
good catch 🤦